kubeluigi/k8s.py

```diff
@@ -345,20 +324,25 @@ def run_and_track_job(k8s_client: ApiClient, job: V1Job) -> None:
     """
     Tracks the execution of a job by following its state changes.
     """
-    logger.info(f"JOB: {job.metadata.name} submitting job")
+    logger.debug(f"JOB: {job.metadata.name} submitting job")
     api_response = k8s_client.create_namespaced_job(
         body=job, namespace=job.metadata.namespace
     )
-    logger.info(f"JOB: {job.metadata.name} submitted")
+    logger.debug(f"JOB: {job.metadata.name} submitted")
     logger.debug(f"API response job creation: {api_response}")
-    logger.info("---Job Definition----")
-    logger.info(job)
-    logger.info("---End of Job Definition")
+    logger.debug("---Job Definition----")
+    logger.debug(job)
+    logger.debug("---End of Job Definition")
     job_completed = False
     with BackgroundJobLogger(job):
         while not has_job_started(job):
             sleep(DEFAULT_POLL_INTERVAL)
-            logger.info("Waiting for Kubernetes job " + job.metadata.name + " to start")
+            logger.debug(
+                "Waiting for Kubernetes job " + job.metadata.name + " to start"
+            )
+
+        pods = get_job_pods(job)
+        yield pods
```

Comment:
so this would work as follows
```python
pods = something. run_and_track_job(..)
generate_links(next(pods)) 
# this will actually yield the pods
# then continue the execution of `run_and_track_job`
# after pods finish running
# then we will call genereate_links
# we print the links after pods have ran
```

Reply:
💡 two ideas stathis :

1. since we have a function that returns us a job definition ,we can use it to generate the logs links before the job actually starts running. Please note that the query only requires the name of the job and we have that before the job starts running on k8s.

2. maybe we can wait for pods to start tunning (like it does in the current code segment) and instead of linking the kuesto query, we can link to the insights of the pod on the azure portal